src/cursor.cpp

```diff
@@ -0,0 +1,46 @@
+#include "util.h"
+#include "voxel.cpp"
+
+class Cursor {
+  private:
+    tuple3i position;
+    tuple3i color;
+    int shape;
+
+  public:
+    void setShape (int flag) {
+      shape = flag;
+    }
+
+    tuple3i getPosition () {
+      return position;
+    }
+
+    tuple3i getColor () {
+      return color;
+    }
+
+    int getShape () {
+      return shape;
+    }
+
+    void setColor (tuple3i newColor) {
+      get<0>(color) = get<0>(newColor);
```

Comment:
Makes sense, I guess I just used it for the symmetry with move(). Probably a good idea to change it to just replace the tuple3i. I'll do that now